crates/core/src/rlp/decode.rs

```diff
@@ -375,6 +375,71 @@ pub fn decode_rlp_item(data: &[u8]) -> Result<(bool, &[u8], &[u8]), RLPDecodeErr
     }
 }
 
+/// Splits an RLP item in two:
+/// - The first item including its prefix
+/// - The remaining bytes after the item
+/// It returns a 3-element tuple with the following elements:
+/// - A boolean indicating if the item is a list or not.
+/// - The payload of the item, including its prefix.
+/// - The remaining bytes after the item.
+pub fn get_item_with_prefix(data: &[u8]) -> Result<(bool, &[u8], &[u8]), RLPDecodeError> {
+    if data.is_empty() {
+        return Err(RLPDecodeError::InvalidLength);
+    }
+
+    let first_byte = data[0];
+
+    match first_byte {
+        0..=0x7F => Ok((false, &data[..1], &data[1..])),
+        0x80..=0xB7 => {
+            let length = (first_byte - 0x80) as usize;
+            if data.len() < length + 1 {
+                return Err(RLPDecodeError::InvalidLength);
+            }
+            Ok((false, &data[0..length + 1], &data[length + 1..]))
```

Comment:
No need to specify the range starts at 0.

Reply:
Done!